### PR TITLE
The Hounds of Tindalos

### PR DESCRIPTION
**Pull request by John-Candlebury**
#### Summary

```SUMMARY: Content "Adds the Hounds of Tindalos, tough Nether monsters that can appear from teleglow"```

#### Purpose of change

More Nasty  and thematic nether creatures. This are about as nasty as you would expect a Lovecraftian horror to be, so be careful.

Also meet Hub 01 fourth mission bossfight,

#### Describe the solution

Much like the creatures from the short story, the Hound of TIndalos are horrid atemporal monsters that prowl the spaces between dimensions and relentlessly pursue unfortunate "Nether-nauts?". AKA they can find you if you have high levels of teleglow. Like the monsters from the books they spawn from corners, and can travel at will into them. They can also desynch and spawn temporal copies of themselves, which are the same monster, with the addition it will dissapear after awhile.

The corner spawning is dictated by an unique fast-decaying field. This field is also spawned with lower intensity when the monsters corner teleport. So be careful, because if a Hound and its copy teleport to the same corner, the resulting field can spawn even more Hounds! Oh joy!

#### Additional context

Unluckily there are corners basically everywhere, so If you meet them early on I suggest you run into an open field and outrun them there. You can also try to break LOS, but be aware that they have very good vision.

Dont use cars because those have corners, and they will teleport into them.

### CODE REVIEW

File: data/json/monsters/monsters.json [hunk 1]
```diff
@@ -1804,6 +1804,92 @@
     "death_function": [ "NORMAL" ],
     "flags": [ "SEES", "HEARS", "SMELLS", "WARM", "BASHES", "GROUP_BASH", "BLEED", "HUMAN", "REVIVES", "CLIMBS", "FILTHY" ]
   },
+  {
+    "id": "mon_hound_tindalos",
+    "type": "MONSTER",
+    "name": "hound of tindalos",
+    "description": "A grotesque hound-like beast, its frame angular and emaciated and its head like that of an abyssal fish.  When it moves, twisted limbs duplicate, merge, disappear, and reform in dissonance, as if a hundred copies of itself where somehow forced to inhabit the same space.  Extraneous to our reality, it moves between the corners of the world, hunting those that would dare stepping beyond time and space.",
```
Comment (Night-Pryanik) [inline data/json/monsters/monsters.json]:
```suggestion
    "description": "A grotesque hound-like beast, its frame is angular and emaciated and its head is like that of an abyssal fish.  When it moves, twisted limbs duplicate, merge, disappear, and reform in dissonance, as if a hundred copies of itself were somehow forced to inhabit the same space.  Extraneous to our reality, it moves between the corners of the world, hunting those that would dare stepping beyond time and space.",
```

File: data/json/monsters/monsters.json [hunk 2]
```diff
@@ -1804,6 +1804,92 @@
...
+      "ARTHROPOD_BLOOD",
+      "NO_BREATHE",
+      "HARDTOSHOOT"
+    ]
+  },
+  {
+    "id": "mon_hound_tindalos_afterimage",
+    "type": "MONSTER",
+    "name": "hound of tindalos",
+    "description": "A grotesque hound-like beast, its frame angular and emaciated and its head like that of an abyssal fish.  When it moves, twisted limbs duplicate, merge, disappear, and reform in dissonance, as if a hundred copies of itself where somehow forced to inhabit the same space.  Extraneous to our reality, it moves between the corners of the world, hunting those that would dare stepping beyond time and space.",
```
Comment (Night-Pryanik) [inline data/json/monsters/monsters.json]:
```suggestion
    "description": "A grotesque hound-like beast, its frame is angular and emaciated and its head is like that of an abyssal fish.  When it moves, twisted limbs duplicate, merge, disappear, and reform in dissonance, as if a hundred copies of itself were somehow forced to inhabit the same space.  Extraneous to our reality, it moves between the corners of the world, hunting those that would dare stepping beyond time and space.",
```

File: src/map.h [hunk 3]
```diff
@@ -1215,6 +1215,9 @@ class map
          */
         void process_falling();
 
+        bool is_cornerfloor( const tripoint &p ) const;
+
+
```
Comment (Night-Pryanik) [inline src/map.h]:
```suggestion
```

File: src/monattack.cpp [hunk 4]
```diff
@@ -4698,6 +4699,58 @@ bool mattack::riotbot( monster *z )
...
+                    // Not teleporting if it means losing sight of our current target
+                    if( z->sees( *target ) ) {
+                        g->m.add_field( dest, fd_tindalos_rift, 2 );
+                        if( g->u.sees( *z ) ) {
+                            add_msg( m_bad, _( "The %s dissipates and reforms close by." ), z->name() );
+                        }
+                        return true;
+                    }
+                }
+
```
Comment (Night-Pryanik) [inline src/monattack.cpp]:
```suggestion
```

File: src/monattack.cpp [hunk 5]
```diff
@@ -4698,6 +4699,58 @@ bool mattack::riotbot( monster *z )
...
+            }
+        }
+        // couldnt teleport without losing sight of target
```
Comment (Night-Pryanik) [inline src/monattack.cpp]:
```suggestion
        // couldn't teleport without losing sight of target
```

**Comment by Night-Pryanik** [comment]
Any chance of some video or a gif to demonstrate its abilities?

File: src/monattack.cpp [hunk 6]
```diff
@@ -4698,6 +4699,58 @@ bool mattack::riotbot( monster *z )
...
+        }
+        if( !free.empty() ) {
+            z->moves -= 140;
+            const tripoint target = random_entry( free );
+            if( monster *const afterimage = g->summon_mon( mon_hound_tindalos_afterimage, target ) ) {
+                afterimage->make_ally( *z );
+            }
+            if( g->u.sees( *z ) ) {
+                add_msg( m_warning,
+                         _( "The hound movements chaotically rewind as a living afterimage splits from it!" ) );
```
Comment (Rivet-the-Zombie) [inline src/monattack.cpp]:
```suggestion
                         _( "The hound's movements chaotically rewind as a living afterimage splits from it!" ) );
```

File: src/player_hardcoded_effects.cpp [hunk 7]
```diff
@@ -622,6 +623,20 @@ void player::hardcoded_effects( effect &it )
                     it.set_duration( 0_turns );
                 }
             }
+            if( one_in( 7200 - ( dur - 360_minutes ) / 4_turns ) ) {
+                add_msg_if_player( m_bad, _( "You are beset with a vision of a prowling beast." ) );
+                for( const tripoint &dest : g->m.points_in_radius( pos(), 6 ) ) {
+                    if( g->m.is_cornerfloor( dest ) ) {
+                        g->m.add_field( dest, fd_tindalos_rift, 3 );
+                        add_msg( m_info, _( "Your surroundings are permeated with a foul scent." ) );
```
Comment (BevapDin) [inline src/player_hardcoded_effects.cpp]:
```suggestion
                        add_msg_if_player( m_info, _( "Your surroundings are permeated with a foul scent." ) );
```
To be in line with the message above.

**Comment by John-Candlebury** [comment]
Heres a gif of a hound chasing a player within a microlab.


tindalos chase

Sorry for the awful quality I had to make it in a rush

**Comment by Rivet-the-Zombie** [comment]
I **love** this.

**Pull request merged**